scripts/lint_jjb.py

```diff
@@ -89,6 +96,67 @@ def parse_file_name(in_dir, in_file):
     return 0
 
 
+# Ensure all jobs have a retention policy
+# Structure:
+#   job:
+#     properties:
+#      - build-discarder:
+#          days-to-keep: 3
+
+def check_retention_value(job, value_name, vmin, vmax):
+    value = jmespath.search(
+        'properties[*]."build-discarder"."{vn}"'.format(vn=value_name), job)[0]
+
+    # try and convert to int, if that fails, check for templated value.
+    try:
+        days = int(value)
+    except ValueError as e:
+        # Can't check templated values, assume they are ok.
+        if '{' in value and '}' in value:
+            return
+        else:
+            raise e
+
+    if days <= vmin or days > vmax:
+        raise ValueError("Value for {vn} out of range. "
+                         "Value: {v}, min: {min}, max: {max}"
+                         .format(vn=value_name, v=value,
+                                 min=vmin, max=vmax))
+
+
+def check_retention(job, in_file):
+    error_message = ("{f}/{j}: Valid build-discarder config not found."
+                     " Please ensure days-to-keep (0 < x <= 365) or"
+                     " num-to-keep (0 < x <= 100) is specified."
+                     " Properties: {p}\n"
+                     .format(j=job['name'], f=in_file,
+                             p=job.get('properties')))
+
+    error_counter = 0
+
```

Comment:
A for-loop might be nicer here.

```python
checks = (
    ('days-to-keep', 0, 365),
    ('num-to-keep', 0, 100),
)

for (retention_type, min_retention, max_retention) in checks:
    try:
        check_retention_value(job, retention_type, min_retention, max_retention)
    except (TypeError, IndexError):
        return_value = 1
    except ValueError as e:
        return_value = 1
        print e
    else:
        return_value = 0
        break
else:
    sys.stderr.write(error_message)

return return_value
```

Reply:
Much better thanks, shortcuts the second check if the first check is successful.